Lib/ufo2ft/makeotfParts.py

```diff
@@ -19,13 +21,17 @@ class FeatureOTFCompiler(object):
     mtiLib into that respective table.
     """
 
-    def __init__(self, font, outline, kernWriter, markWriter, mtiFeaFiles=None):
+    def __init__(self, font, outline, kernWriterClass=KernFeatureWriter,
+                 markWriterClass=MarkFeatureWriter, mtiFeaFiles=None):
         self.font = font
         self.outline = outline
-        self.kernWriter = kernWriter
-        self.markWriter = markWriter
+        self.kernWriterClass = kernWriterClass
+        self.markWriterClass = markWriterClass
         self.mtiFeaFiles = mtiFeaFiles
-        self.setupAnchorPairs()
+        #
+        self.anchorPairs = []
+        self.ligaAnchorPairs = []
+        self.mkmkAnchorPairs = []
```

Comment:
The primary motivation of this PR is to make feature compilation steps more self-contained, e.g. not setup anchor pairs unless the markWriter is called or make writers optional.